core-bundle/src/Filesystem/FilesystemItem.php

```diff
@@ -106,9 +108,31 @@ public function withPath(string $path): self
             $this->fileSize,
             $this->mimeType,
             $this->extraMetadata,
+            $this->storage,
         );
     }
 
+    public function withStorage(VirtualFilesystemInterface $storage): self
+    {
+        return new self(
+            $this->isFile,
+            $this->path,
+            $this->lastModified,
+            $this->fileSize,
+            $this->mimeType,
+            $this->extraMetadata,
+            $storage,
+        );
+    }
+
+    /**
+     * @internal
+     */
+    public function getStorage(): VirtualFilesystemInterface
```

Comment:
Nitpick: Maybe add a `hasStorage(): bool` on top to check for that stuff in the template or so? Without getting an exception.

Reply:
For filesystem items getting to templates/public API, these will always have a storage set. Internally, the DBAFS' and the `MountManager` also create items  when listing elements - these do not have a storage but are enriched to have one when accessed via one. So that's why it's not a mandatory constructor argument.